requirements-min.txt

```diff
@@ -1,7 +1,8 @@
 # Minimum package versions that ONNX supports
 protobuf==3.20.2; python_version<"3.12"
 protobuf==4.25.1; python_version>="3.12"
-numpy==1.20.3; python_version<"3.10"
+numpy==1.21.0; python_version<"3.10"
 numpy==1.23.2; python_version=="3.10"
 numpy==1.23.2; python_version=="3.11"
 numpy==1.26.0; python_version>="3.12"
+pillow<10.4
```

Comment:
A couple of notes:
* Since we are increasing the minimum numpy, do we still need to explicitly specify pillow or should we go back to leaving it implicit? The issue in pillow 10.4 seemed related to numpy 1.20.
* If we do specify pillow, should we do `pillow!=10.4`? Theoretically pillow's https://github.com/python-pillow/Pillow/pull/8187 should fix the numpy 1.20 issue in their next release
* Should the changes to the requirements files merge into `main` first and then be cherry picked into a 1.16.2 branch?
    * The yml file changes appear to be 1.16.2 specific as 1.17.0 and beyond will support numpy 2. 
    * The changes in the requirements regarding pillow would likely still apply to 1.17.0 (except the numpy<2 in requirements-dev.txt)